### PR TITLE
Remove superfluous CpGridData::allcorners and save space during setup

### PR DESCRIPTION
**Pull request by blattms**

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
jenkins run norne

**Comment by blattms** [comment]
Norne run through.  Any comments?
Otherwise I will join the happy self-merge-community. :speak_no_evil:

**Comment by atgeirr** [comment]
I find it a bit hard to see if there are any changes to the code that has been moved around (such as the dim-0 geometry specialization), are they just moved or also changed?

**Comment by akva2** [comment]
note that norne did NOT run cause somebody forgot their manners.

**Comment by atgeirr** [comment]
jenkins run norne please

**Comment by blattms** [comment]
I am sorry about moving things, but it was needed to make the code compile.
Otherwise my compiler would have complained about missing
`Geometry<0,3>::center()` method.
The `Geometry<0,cdim>` is unchanged. Only `Geometry<3,cdim>` now uses
`Geometry<0,cdim>` instead of the allcorners_

**Comment by atgeirr** [comment]
> I am sorry about moving things, but it was needed to make the code compile.

No problem, I just wanted to know if there were hard-to-notice changes. I'll do a review now.

File: opm/grid/cpgrid/CpGridData.cpp [hunk 1]
```diff
@@ -878,6 +814,64 @@ void CpGridData::distributeGlobalGrid(const CpGrid& grid,
 
     logical_cartesian_size_=view_data.logical_cartesian_size_;
 
+        // Set up the new topology arrays
```
Comment (atgeirr) [inline opm/grid/cpgrid/CpGridData.cpp]: Nitpick: indentation.

File: opm/grid/cpgrid/CpGridData.cpp [hunk 2]
```diff
@@ -878,6 +814,64 @@ void CpGridData::distributeGlobalGrid(const CpGrid& grid,
...
+    // Now copy the point geometries that do exist.
+    for(auto begin=point_indicator.begin(), pi=begin,  end=point_indicator.end(); pi!=end;
+        ++pi)
+    {
+        if(*pi<std::numeric_limits<int>::max())
+        {
+            point_geom.emplace_back(global_point_geom[pi-begin]);
+        }
+    }
+
```
Comment (atgeirr) [inline opm/grid/cpgrid/CpGridData.cpp]: Nitpick: extra blank

File: opm/grid/cpgrid/CpGridData.cpp [hunk 3]
```diff
@@ -878,6 +814,64 @@ void CpGridData::distributeGlobalGrid(const CpGrid& grid,
...
     logical_cartesian_size_=view_data.logical_cartesian_size_;
 
+        // Set up the new topology arrays
+    // Count the existing points and allocate space
+    EntityVariable<cpgrid::Geometry<0, 3>, 3>& point_geom = geometry_.geomVector(std::integral_constant<int,3>());
+    const std::vector<cpgrid::Geometry<0, 3> >& global_point_geom=view_data.geomVector<3>();
+    point_geom.reserve(noExistingPoints);
+
+    // Now copy the point geometries that do exist.
+    for(auto begin=point_indicator.begin(), pi=begin,  end=point_indicator.end(); pi!=end;
```
Comment (atgeirr) [inline opm/grid/cpgrid/CpGridData.cpp]: Nitpick: space after `for` and around `=`.

File: opm/grid/cpgrid/EntityRep.hpp [hunk 4]
```diff
@@ -213,6 +213,7 @@ namespace Dune
         class EntityVariableBase : private std::vector<T>
         {
             friend class CpGridData;
+            using std::vector<T>::operator[];
```
Comment (atgeirr) [inline opm/grid/cpgrid/EntityRep.hpp]: Isn't this superfluous?

File: tests/cpgrid/geometry_test.cpp [hunk 5]
```diff
@@ -116,8 +117,13 @@ BOOST_AUTO_TEST_CASE(cellgeom)
 //     for (int i = 0; i < 8; ++i) {
 //         std::cout << corners[i] << std::endl;
 //     }
+    cpgrid::EntityVariable<cpgrid::Geometry<0, 3>, 3> pg;
+    pg.reserve(8);
+    for(const auto& c : corners)
+        pg.push_back(cpgrid::Geometry<0, 3>(c));
```
Comment (atgeirr) [inline tests/cpgrid/geometry_test.cpp]: Nitpick: braces.

File: opm/grid/cpgrid/CpGridData.cpp (hunk 2 above)
Comment (blattms) [reply opm/grid/cpgrid/CpGridData.cpp]: I have no idea what you mean by that. Extra line?

File: opm/grid/cpgrid/EntityRep.hpp (hunk 4 above)
Comment (blattms) [reply opm/grid/cpgrid/EntityRep.hpp]: Indeed. it did not work as expected as my compiler always choose the one taking the Entity, anyway.

**Comment by blattms** [comment]
Except for the "extra blank" request (that I did not get) all issues should be fixed with the last commit.

File: opm/grid/cpgrid/CpGridData.cpp (hunk 2 above)
Comment (blattms) [reply opm/grid/cpgrid/CpGridData.cpp]: Finally got it. Blank before and after the minus sign. Amended last commit

**Comment by blattms** [comment]
Just to be sure: another test run.

**Comment by blattms** [comment]
jenkins build this please.

File: opm/grid/cpgrid/CpGridData.cpp (hunk 2 above)
Comment (atgeirr) [reply opm/grid/cpgrid/CpGridData.cpp]: Sorry for being too brief, I actually meant extra blank *line*.

**Comment by atgeirr** [comment]
Thanks, merging!

**Comment by akva2** [comment]
this broke building with clang, ref https://ci.opm-project.org/job/opm-simulators-static-analysis/33/console

**Comment by akva2** [comment]
fix in #382

**Pull request merged**